File: harvest-analysis.jl
#############################
#harvest-analysis.jl
#Brandon Edwards
#Last Updated: August 2016
#############################

#############################
# Constants
#############################

#Should be current year - 1
analysisYear = 2015

#############################
# Packages needed
#############################

using DataFrames, Gadfly

#############################
# File IO
#############################
projPath = Base.source_path()
separateDirChar = "\\"
pathEndSearch = rsearch(projPath, separateDirChar)[1]
endOfDirPath = projPath[pathEndSearch:length(projPath)]
projPath = split(projPath, endOfDirPath)[1]
file = string(projPath, "\\", analysisYear, "_Harvest_Data_All.csv")

#############################
# Import Data
#############################

data = readtable(file)
#Drop data with NA for month
data = data[!isna(data[:MONTH]),:]

#############################
# Entire Lake Analysis
#############################
lakeResults = DataFrame(Month = Int[], HarvestWeight = Float64[], Events = Int64[])

for month = 1:12
  tempData = data[data[:MONTH] .== month, :]
  push!(lakeResults, (month, sum(tempData[:HVSWT_KG]), size(tempData)[1]))
end

File: harvest-analysis.R
####################################
#
#	harvest-analysis.R
#	Brandon Edwards
# Last Update: August 2016
#
####################################

####################################
#	Clear Memory
####################################

remove(list=ls())

####################################
#	Constants
####################################

#Should be currentYear - 1
AnalysisYear <- 2015

####################################
# File IO
####################################

mainDir <- paste(getwd(), "/", sep = "")
subDir <- paste("Plots/", sep = "")
dir.create(file.path(mainDir, subDir))

####################################
#	Load Data
####################################	

data <- read.csv(paste(mainDir, AnalysisYear, "_Harvest_Data_All.csv", sep=""))

####################################
#	Analyze Entire Lake
####################################

lake.results <- NULL

for (month in 1:12)
{
  data.temp <- data[data$MONTH %in% c(month), ]
  total.weight <- sum(data.temp$HVSWT_KG)
  lake.results <- rbind(lake.results, (month, total.weight))
}
